tensorflow/java/src/main/java/org/tensorflow/op/AbstractSingleOp.java

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.op;
+
+import org.tensorflow.Operation;
+
+/** A base class for an {@link Op} that wraps a single operation. */
+public abstract class AbstractSingleOp implements Op {
```

Comment:
I see. Makes sense. Though, perhaps a different name. `SingleOp` is a bit confusing, because aren't all `Op` implementations a "single" "op"? :)

I'm guessing you meant to convey the difference between some composite operations and wrappers of "primitive" operations, i.e., operations that have a 1:1 correspondence with a node in the graph? What examples of composite operations did you have in mind? That said, how about `PrimitiveOp` instead? (`AbstractSingleOp` or `SingleOp` is also fine, just proposing an alternative to see if it appeals to you)

Let's also elaborate on the documentation about what "single"/"primitive" mean?

```java
/** Base class for {@link Op} implementations that are backed by a single {@link org.tensorflow.Operation}. */
```